### PR TITLE
button: improve a11y

### LINKED ISSUE #1479: button: improve JAWS readability
JAWS should read
- [x] `button` as "button"
- [x] `button_type_link` as "link"
- [x] `button_type_submit` as "button"
- [x] `button_togglable_check` as "toggle button"
- [x] `button_togglable_radio` as "toggle button"
- [x] `_disabled` as "unavailable"
- [x] `button_checked` with specified `_togglable` modifier as "toggle button pressed"
- [x] `text` special field
- [x] `title` special field

Check and fix divergencies.

### PR DESCRIPTION
**Pull request by sipayRT**
Closes #1479

### CODE REVIEW

**Comment by aristov** [comment]
@sipayRT does it wait for #1567?

**Comment by sipayRT** [comment]
@aristov no, #1567 need just for attach block.

File: common.blocks/button/_togglable/button_togglable.bh.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function(bh) {
+    bh.match(['button_togglable_check', 'button_togglable_radio'], function(ctx) {
+        ctx
+            .applyBase()
```
Comment (aristov) [inline common.blocks/button/_togglable/button_togglable.bh.js]: Is it necessary to call `applyBase` here?

File: common.blocks/button/_togglable/button_togglable.bh.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+            .attr('aria-pressed', ctx.mod('checked') || false);
```
Comment (aristov) [inline common.blocks/button/_togglable/button_togglable.bh.js]: `!!ctx.mod('checked')`

File: common.blocks/button/_togglable/button_togglable.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * @exports
+ * @class button
+ * @bem
+ */
+provide(Button.decl({ modName : 'togglable' }, /** @lends button.prototype */{
+    onSetMod : {
+        'checked' : function(_, modVal) {
+            this.__base.apply(this, arguments);
+            this.domElem.attr('aria-pressed', modVal || false);
```
Comment (aristov) [inline common.blocks/button/_togglable/button_togglable.js]: `!!modVal`

File: common.blocks/button/_togglable/button_togglable_check.bemhtml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+block('button').mod('togglable', 'check').attrs()(function() {
+    return this.extend(applyNext(), { 'aria-pressed' : this.mods.checked || false });
```
Comment (aristov) [inline common.blocks/button/_togglable/button_togglable_check.bemhtml]: `!!this.mods.checked`

File: common.blocks/button/_type/button_type_link.bemhtml [hunk 5]
```diff
@@ -3,7 +3,9 @@ block('button').mod('type', 'link')(
 
     attrs()(function() {
         var ctx = this.ctx,
-            attrs = {};
+            attrs = {
+                role : 'link'
```
Comment (aristov) [inline common.blocks/button/_type/button_type_link.bemhtml]: Reduce to single line object.

File: common.blocks/button/button.js [hunk 6]
```diff
@@ -38,6 +38,10 @@ provide(BEMDOM.decl({ block : this.name, baseBlock : Control }, /** @lends butto
         },
 
         'disabled' : {
+            '*' : function(_, modVal) {
+                this.__base.apply(this, arguments);
+                this.domElem.attr('aria-disabled', !!modVal);
```
Comment (aristov) [inline common.blocks/button/button.js]: Omit `aria-disabled="false"`.

File: common.blocks/button/button.spec.js [hunk 7]
```diff
@@ -83,6 +83,16 @@ describe('button', function() {
...
         });
+
+        it('should update "aria-disabled" attr on set "disabled" mod', function () {
+            button
+                .setMod('disabled')
+                .domElem.attr('aria-disabled').should.be.equal('true');
+
+            button
+                .delMod('disabled')
+                .domElem.attr('aria-disabled').should.be.equal('false');
```
Comment (aristov) [inline common.blocks/button/button.spec.js]: Expect to be `undefined`

File: common.blocks/button/_togglable/button_togglable_radio.bemhtml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+block('button').mod('togglable', 'radio').attrs()(function() {
+    return this.extend(applyNext(), { 'aria-pressed' : this.mods.checked || false });
```
Comment (aristov) [inline common.blocks/button/_togglable/button_togglable_radio.bemhtml]: `!!this.mods.checked`

File: common.blocks/button/_togglable/button_togglable.bh.js (hunk 1 above)
Comment (sipayRT) [inline common.blocks/button/_togglable/button_togglable.bh.js]: tnx for your attention. I'll remove it

**Comment by sipayRT** [comment]
@aristov updated

**Comment by aristov** [comment]
LGTM

**Pull request merged**